documentation/TODO.md

```diff
@@ -0,0 +1,166 @@
+* Split each store documentation into its own distinct .adoc file
+* All cache store docs should be as sub-chapters of the main persistence chapter
+* Details about store internals should go after documentation about configuration and types of stores. They are of interest mostly to store developers
+
+### Clustering (Dan)
+* Transactions (Pedro)
+* The Total Order docs should be moved under here
+
+## Locking and concurrency (Pedro)
+* The Versioning paragraph mentions tombstones as if they exist.
+
+### Local mode cache (Radim)
+* The simple cache documentation is erroneously included in the functional map API chapter. Move it here
+* Update the information inside the performance paragraph to include results from the simple cache benchmarks
+
+### Query + Indexing (Adrian & Gustavo)
+* Add missing parts (CQ)
+
+### Consistency guarantees (Dan)
+* Refactor the WIKI page into the documentation, possibly as its own chapter.
+
+### Streams (Will)
+* In the serialization section mention the implicit overloading of lambdas to serializable
+* Also provide an example of how one would implement an AdvancedExternalizer
+
+### Map/Reduce (Vladimir)
+* This feature no longer exists, it needs to go.
+
+### CDI (Sebastian)
+* Move under an overarching Integrations chapter
+* Make sure we mention split artifacts (embedded and remote)
+* Prevent mentioning configuration overriding 2 times (one for embedded and one for remote)
+
+### Spring (Sebastian)
+* Move under an overarching Integrations chapter
+* Move to its own paragraph under Integrations
+* Do not mention EHCache… it’s not relevant
+* Mention proper artifacts (spring4)
+
+### Remote Protocols (Tristan)
+* The chapter is currently called “Server modules” but this is confusing as we are really discussing the remoting protocols and neither the WildFly modules nor Infinispan Server
+* This should be turned upside down, looking at the protocol side first, and then detailing the server-side implementation aspects.
+
+### Management Tooling (Vittorio)
+* Remove the RHQ section and other mentions of it
+* Move the CLI chapter under here
+
+### Custom Interceptors (Dan)
+* Since we are describing an advanced API this needs to be moved further up
+* Document the new async design and provide some useful examples
+
+### Running on AWS (Sebastian + Gustavo)
+* Actually refactor this chapter into a more general chapter about running in different cloud/container environments
+* AWS
+* Docker
+* OpenShift / Kubernetes
+* Google Compute Engine
+* Azure
+* ...
+
+### WildFly Modules (Gustavo)
+* Move under an overarching Integrations chapter
+
+### Default values for properties (Ryan)
+* Remove this chapter and find a way to generate the information into the configuration reference
+
+### HTTP session clustering and caching (Sebastian)
+* Move under an overarching Integrations chapter
+* Remove the  open tasks
+* Mention Spring Session
+* Eventually will also contain Paul’s Tomcat adapter
+
+### Hadoop + Spark
```

Comment:
How about link to https://github.com/infinispan/infinispan-spark/blob/master/README.md and  https://github.com/infinispan/infinispan-hadoop/blob/master/README.md ?